javascript/ql/src/semmle/javascript/Regexp.qll

```diff
@@ -811,7 +821,11 @@ predicate isInterpretedAsRegExp(DataFlow::Node source) {
     exists(MethodCallExpr mce, string methodName |
       mce.getReceiver().analyze().getAType() = TTString() and
       mce.getMethodName() = methodName and
-      not exists(DataFlow::FunctionNode func | func = DataFlow::valueNode(mce.getCallee()).getAFunctionValue() | not func.getFunction().inExternsFile())
+      not exists(Function func |
+        func = any(DataFlow::MethodCallNode call | call.getEnclosingExpr() = mce).getACallee()
```

Comment:
Can we avoid the `getEnclosingExpr` dance by making `mce` a `MethodCallNode` straight away and rephrasing the code relating it to `source` below?